### ISSUE TITLE
vizualizacija

### ISSUE THREAD

**Issue by VidStarc** [VidStarc/APPR-2014-15#3]
Pozdravljeni!
jaz imam podatke za zahodno in vzhodno Slovenijo in to želim tudi predstaviti na zemljevidu. Nikjer pa nisem našel takega zemljevida, tako da sem pobral kar zemljevid Slovenije po regijah, ter jih nato razvrstil glede na to ali spadajo v zahodno ali vzhodno Slovenijo. Zataknilo pa se mi je že pri tem ko moram uporabiti funkcijo preuredi. Namreč želim prikazati skupno povprečje(v zahodni in vzhodni Sloveniji), a nevem kako bi to storil.
hvala in lp, Vid Starc

**Comment by jaanos** [comment]
Ta rešitev z regijami je v redu - je pa res, da na samo dveh regijah ne moreš veliko prikazati. Mogoče bi bilo smiselno, da narišeš več zemljevidov (npr. skozi leta), da bodo vidne spremembe skozi čas.

Funkcija `preuredi` tukaj ne bo prišla v upoštev - ta je namenjena urejanju, ko nimaš popolnega ujemanja med imeni v podatkih in v zemljevidu. Namesto tega lahko narediš npr. nekaj takega:
```R
zahod <- c(1,2,6,7)
stolpci <- 6:9
govedo.zahod <- stevilo.goveda[45, stolpci]
govedo.vzhod <- stevilo.goveda[23, stolpci]
norm.zahod <- (govedo.zahod-min(govedo.zahod)*0.99)/(max(govedo.zahod)-min(govedo.zahod)*0.99)
norm.vzhod <- (govedo.vzhod-min(govedo.vzhod)*0.99)/(max(govedo.vzhod)-min(govedo.vzhod)*0.99)

# Rišemo zemljevide v mrežo velikosti 2x2
par(mfrow=c(2,2))
for (i in 1:4) {
  plot(slo, col=ifelse(1:nrow(slo) %in% zahod,
                       rgb(1, 0, 0, norm.zahod[[i]]),
                       rgb(0, 0, 1, norm.vzhod[[i]])))
  title(paste("Leto", 2009+i))
}
# Povrnemo prvotno nastavitev
par(mfrow=c(1, 1))
```
Tako bodo za vsako regijo posebej vidne spremembe skozi čas. Primerjava med regijama tukaj ne pride v upoštev, ker so razlike prevelike. Če bi hotel direktno primerjati, potem bi moral upoštevati še npr. populacijo ali površino vsake regije.

Namesto tega bi po mojem bilo boljše, če pridobiš kakšne druge podatke (npr. za evropske države), pa bo potem lažje narisati bolj pomenljiv zemljevid.

**Comment by VidStarc** [comment]
ok hvala, točno to sem hotel. Sedaj pa poskušam še podatke za evropsko unijo naložit pa mi nikakor ne uspe uvozit iz Eurostata (tudi na githubu lahko vidite da mi datoteko stevilo-goveda.csv odpre kot tabelo namesto kot csv, torej besede ločene z vejico). morda veste zakaj to dela?

**Comment by VidStarc** [comment]
popravek: stevilo-govedaEU.csv odpre kot tabelo

**Comment by jaanos** [comment]
Datoteka CSV ni pravilno oblikovana. Sklepam, da si jo odprl v Excelu in jo potem shranil. Namesto tega jo neposredno shrani iz brskalnika. Seveda bo potrebno še popraviti sam uvoz, saj imaš tukaj drugačne stolpce.

**Comment by VidStarc** [comment]
ne, to je direktno iz strani Eurostat (download as CSV file)

**Comment by jaanos** [comment]
Tu gre najbrž za podatke s spletne strani

http://appsso.eurostat.ec.europa.eu/nui/show.do?dataset=apro_mt_lscatl&lang=en

Če izberem _Download_ in nato _Download in CSV Format_, dobim ZIP, kjer je notri pravilno oblikovana datoteka. To si skopiraj med podatke in je ne odpiraj v Excelu. Pred prenosom lahko odkljukaš _Flags and footnotes_; števila bo potrebno še ustrezno pretvoriti po uvozu:
```R
stevilo.govedaEU$Value <- as.numeric(gsub(",", "", stevilo.govedaEU$Value))
```

**Comment by VidStarc** [comment]
Ja, naredim to kar ste napisali, pa mi vseeno ne vrže tako tabelo kot je na Eurostatu, vrže mi tako:
2007 Belgium
2008 Belgium
   .
   .
Potem pa sem se sprijaznil s tem in sem težavo rešil drugače ko sem risal zemljevid. Sedaj pa mi noče narisat zemljevida in javlja napako: 'from' cannot be NA, NaN or infinite . Nevem kako to popraviti.
